packages/dataviews/src/components/dataform-context/index.tsx

```diff
@@ -0,0 +1,53 @@
+/**
+ * WordPress dependencies
+ */
+import { createContext, useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type { FormField, NormalizedField } from '../../types';
+
+type DataFormContextType< Item > = {
+	getFieldDefinition: (
+		field: FormField | string
+	) => NormalizedField< Item > | undefined;
+};
+
+const DataFormContext = createContext< DataFormContextType< any > >( {
+	getFieldDefinition: () => undefined,
+} );
+
+export function DataFormProvider< Item >( {
+	fields,
+	children,
+}: React.PropsWithChildren< { fields: NormalizedField< Item >[] } > ) {
+	const getFieldDefinition = useCallback(
+		( field: FormField | string ) => {
+			const fieldId = typeof field === 'string' ? field : field.id;
+
+			const definition = fields.find(
+				( fieldDefinition ) => fieldDefinition.id === fieldId
+			);
+			if ( definition ) {
+				return {
+					...definition,
+					label:
```

Comment:
This is weird to me, why are we overriding the definition. What "type" are we returning here.

Reply:
We are returning the `NormalizedField` type here. I explained the reasoning for going this route as part of this comment: https://github.com/WordPress/gutenberg/pull/66531#discussion_r1824481795
But with [this change](https://github.com/WordPress/gutenberg/pull/66531#discussion_r1829334630), some of the `NormalizedField` retrieval logic will move back to the layout, so I can keep it there. Which would be better to avoid confusion.

Reply:
I removed this.